### ISSUE TITLE
[Suggestion] In game chat

### ISSUE THREAD

**Issue by Sullos** [tsunamayo/Starship-EVO#1829]
Hey Tsuna, now I'm sorry if this has been suggested already, I'm sure it came up at some point...
But if you are pushing for multiplayer gameplay as being a selling point upon Early Access Release, my suggestion is that we need a very basic in-game chat system.

As an Idea of how other games do it
Minecraft uses the: Press T to open the chat box, that lets you type until you either send the message with enter, or press escape to cancel typing.
Most other games are similar as chat is a fairly straightforward system, but as for keys that are used, some games use Enter, others use T (for talk I imagine). And then various other keys.

You already have a player name system in-game, where the player sets their in-game name. I suggest you just use that as a player Identity for the message sent by that player and just have it as Player 1 or Player 2, etc for every person who hasn't set their name.

Of course this all depends on how much you're focusing on the multiplayer aspect of the game :)
Thanks Tsuna

**Comment by ZachZent** [comment]
Also a way to see everyone in the current server

**Comment by tsunamayo** [comment]
Yes I have many quality of life improvement to do. Problem being I am short on time, and also I still need to do the big technological switch as my current network tech is being deprecated by Unity. So it bugs me to implement some feature with the old tech knowing I would have to migrate them... For the chat it will likely come after ea, for now the goal is more to play with friends, and I kinda assume that people talk on mic (I guess?...)

**Comment by Vincent14** [comment]
It's not always possible to get friends on mic, and sharing in-game text could help when you can't spell something clearly with you voice (especially for international communities).

Could be linked to this issue https://github.com/tsunamayo/Starship-EVO/issues/3300 => Player coordinates is a must have (with the Minecraft overlay in mind)
